File: Problemset/middle-of-the-linked-list/middle-of-the-linked-list.py

# @Title: 链表的中间结点 (Middle of the Linked List)
# @Author: 15218859676
# @Date: 2020-08-19 16:21:20
# @Runtime: 12 ms
# @Memory: 12.9 MB

# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def middleNode(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        fast = head 
        slow = head 
        while  fast and fast.next:
            fast = fast.next.next
            slow = slow.next
        return slow

File: Problemset/middle-of-the-linked-list/middle-of-the-linked-list.java

// @Title: 链表的中间结点 (Middle of the Linked List)
// @Author: 15218859676
// @Date: 2020-08-19 16:19:41
// @Runtime: 0 ms
// @Memory: 37 MB

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        // ListNode fast = head.next;
        // ListNode slow = head;
        // while(fast!=null && fast.next!= null){
        //     fast = fast.next.next;
        //     slow = slow.next;
        // }
        // return slow;
        ListNode fast = head;
        ListNode slow = head;
        while(fast != null && fast.next != null){
            fast = fast.next.next;
            slow = slow.next;
        }
        return slow;

    }
}
